machine/dni/dni_3048up/rootconf/sysroot-bin/onie-sysinfo

```diff
@@ -0,0 +1,271 @@
+    fi
+
+}
+
+[ -r $lib_dir/sysinfo-arch ]     && . $lib_dir/sysinfo-arch
+[ -r $lib_dir/sysinfo-platform ] && . $lib_dir/sysinfo-platform
+
+args="hsSevimrpcfdatP"
+
+usage()
+{
+    echo "$(basename $0) [-${args}]"
+    cat <<EOF
+Dump ONIE system information.
+
+COMMAND LINE OPTIONS
+
+	The default is to dump the ONIE platform (-p).
+
+	-h
+		Help.  Print this message.
+
+	-s
+		Serial Number
+
+	-P
+		Part Number
+
+	-e
+		Management Ethernet MAC address
+
+	-v
+		ONIE version string
+
+	-i
+		ONIE vendor ID.  Print the ONIE vendor's IANA enterprise number.
+
+	-m
+		ONIE machine string
+
+	-r
+		ONIE machine revision string
+
+	-p
+		ONIE platform string.  This is the default.
+
+	-c
+		ONIE CPU architecture
+
+	-f
+		ONIE configuration version
+
+	-d
+		ONIE build date
+
+	-t
+		ONIE partition type
+
+	-S
+		ONIE silicon switch vendor
+
+	-a
+		Dump all information.
+EOF
+}
+
+sn=no
+mac=no
+version=no
+vendor_id=no
+machine=no
+machine_rev=no
+platform=no
+pn=no
+arch=no
+config_version=no
+build_date=no
+partition_type=no
+switch_asic=no
+all=no
+
+[ $# -eq 0 ] && platform=yes
+
+while getopts "$args" a ; do
+    case $a in
+        h)
+            usage
+            exit 0
+            ;;
+        s)
+            sn=yes
+            ;;
+        S)
+            switch_asic=yes
+            ;;
+        e)
+            mac=yes
+            ;;
+        v)
+            version=yes
+            ;;
+        i)
+            vendor_id=yes
+            ;;
+        m)
+            machine=yes
+            ;;
+        r)
+            machine_rev=yes
+            ;;
+        p)
+            platform=yes
+            ;;
+        P)
+            pn=yes
+            ;;
+        c)
+            arch=yes
+            ;;
+        f)
+            config_version=yes
+            ;;
+        d)
+            build_date=yes
+            ;;
+        t)
+            partition_type=yes
+            ;;
+        a)
+            all=yes
+            ;;
+        *)
+            echo "Unknown argument: $a"
+            usage
+            exit 1
+    esac
+done
+
+count=1
+print_val()
+{
+    if [ $count -gt 1 ] ; then
+        echo -n " "
+    fi
+    if [ -n "$1" ] ; then
+        echo -n "$1"
+    else
+        echo -n "unknown"
+    fi
+    count=$(( $count + 1 ))
+}
+
+if [ "$sn" = "yes" ] || [ "$all" = "yes" ] ; then
+    val=$(get_serial_num)
+    print_val $val
+fi
+
+if [ "$pn" = "yes" ] || [ "$all" = "yes" ] ; then
+    val=$(get_part_num)
+    print_val $val
+fi
+
+if [ "$mac" = "yes" ] || [ "$all" = "yes" ] ; then
+    val=$(get_ethaddr)
+    print_val $val
+fi
+
+for v in version vendor_id machine machine_rev platform arch config_version partition_type build_date switch_asic ; do
+    eval dump='$'$v
+    if [ "$dump" = "yes" ] || [ "$all" = "yes" ] ; then
+        eval val='$onie'_$v
+        print_val $val
+        if [ "$version" = "yes" ] || [ "$v" = "version" ] ; then
+        	uboot_v="U-Boot-""$(cat /dev/mtd0 | grep '3048UP V' | awk '{print $13}')"
```

Comment:
We had considered about setting the  variable "VENDOR_VERSION"  in machine.make 
shown as below : 

VENDOR_VERSION = -K_1.0.7-$(UBOOT_VERNOR_VERSION)
UBOOT_VERNOR_VERSION = U_1.0.8
...
"onie-sysinfo -v" will display like that:
ONIE:/ # onie-sysinfo -v
2016.05-K_1.0.7-U_1.0.8

But we have three questions:

1. When we update the u-boot version ( modify UBOOT_VERNOR_VERSION  ), we need to              rebuild ONIE kernel image and update ONIE kernel image.
 Will that be OK ?

If we only modify the u-boot next time. We need to modify
 UBOOT_VERNOR_VERSION =  U_1.0.9.
And should  VENDOR_VERSION = -K_1.0.7-$(UBOOT_VERNOR_VERSION) or
                     VENDOR_VERSION = -K_1.0.8-$(UBOOT_VERNOR_VERSION)   ?


2. if we update the u-boot version to other version with u-boot bin file in u-boot CLI.
The command  "onie-sysinfo -v" (VENDOR_VERSION = -K_1.0.7-$(UBOOT_VERNOR_VERSION) may not show the real u-boot version.
Will that be OK ?
 
3. Or should we merge the ONIE kernel version and u-boot version to the same version ?
    such as VENDOR_VERSION = -DNI_V2.0.0 
    once we modify u-boot or kernel, the VENDOR_VERSION will be updated.